src/test/shell/bazel/bazel_java_test_defaults.sh

```diff
@@ -240,4 +240,65 @@ EOF
   expect_not_log "singlejar/singlejar_local"
 }
 
+function test_executable_java_binary_compiles_for_platform_without_cc_toolchain() {
+  mkdir -p pkg
+  cat > pkg/BUILD.bazel <<'EOF'
+# Choose a platform with a registered JDK for which no registered C++ toolchain
```

Comment:
Is there something about (windows, arm64, c++) that ensures this assumption will always hold? If not, this test may start passing incorrectly at some point in the future. Could we maybe check that building a cc target fails on this platform?

Reply:
Good point. I added the check and also construct a fake platform that should make this test more stable as we add support for more platforms.